### ISSUE TITLE
The number of matches between constructs* is incorrect in certain cells.

### ISSUE THREAD

**Issue by sambrhim** [markheckmann/OpenRepGrid.ic#6]
For example, in the matching matrix (below) of the following grid, the matches between C1 and C17 or C18 are given as "4" instead of "0". One can also see a diagonal of "4" which is not correct.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/samiibrahim/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/samiibrahim/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

0 | E4 | E3 | E2 | E1 | 1 | preferred
-- | -- | -- | -- | -- | -- | --
C1+ | 0 | 0 | 0 | 0 | C1- | 0
C2+ | 0 | 0 | 0 | 0 | C2- | 0
C3+ | 0 | 0 | 0 | 1 | C3- | 0
C4+ | 0 | 0 | 1 | 0 | C4- | 0
C5+ | 0 | 0 | 1 | 1 | C5- | 0
C6+ | 0 | 1 | 0 | 0 | C6- | 0
C7+ | 0 | 1 | 0 | 1 | C7- | 0
C8+ | 0 | 1 | 1 | 0 | C8- | 0
C9+ | 0 | 1 | 1 | 1 | C9- | 0
C10+ | 1 | 0 | 0 | 0 | C10- | 0
C11+ | 1 | 0 | 0 | 1 | C11- | 0
C12+ | 1 | 0 | 1 | 0 | C12- | 0
C13+ | 1 | 0 | 1 | 1 | C13- | 0
C14+ | 1 | 1 | 0 | 0 | C14- | 0
C15+ | 1 | 1 | 0 | 1 | C15- | 0
C16+ | 1 | 1 | 1 | 0 | C16- | 0
C17+ | 1 | 1 | 1 | 1 | C17- | 0
C33+ | 1 | 1 | 1 | 1 | C33- | 0

</body>

</html>

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/samiibrahim/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/samiibrahim/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

Number   of matches between constructs* |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
  |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |  
constructs |   | C1 | C2 | C3 | C4 | C5 | C6 | C7 | C8 | C9 | C10 | C11 | C12 | C13 | C14 | C15 | C16 | C17 | C18
C1- - C1+ | C1 |   | 4 | 3 | 3 | 2 | 3 | 2 | 2 | 3 | 3 | 2 | 2 | 3 | 2 | 3 | 3 | 4 | 4
C2- - C2+ | C2 | 4 |   | 3 | 3 | 2 | 3 | 2 | 2 | 3 | 3 | 2 | 2 | 3 | 2 | 3 | 3 | 4 | 4
C3- - C3+ | C3 | 3 | 3 |   | 2 | 3 | 2 | 3 | 3 | 2 | 2 | 3 | 3 | 2 | 3 | 2 | 4 | 3 | 3
C4- - C4+ | C4 | 3 | 3 | 2 |   | 3 | 2 | 3 | 3 | 2 | 2 | 3 | 3 | 2 | 3 | 4 | 2 | 3 | 3
C5- - C5+ | C5 | 2 | 2 | 3 | 3 |   | 3 | 2 | 2 | 3 | 3 | 2 | 2 | 3 | 4 | 3 | 3 | 2 | 2
C6- - C6+ | C6 | 3 | 3 | 2 | 2 | 3 |   | 3 | 3 | 2 | 2 | 3 | 3 | 4 | 3 | 2 | 2 | 3 | 3
C7- - C7+ | C7 | 2 | 2 | 3 | 3 | 2 | 3 |   | 2 | 3 | 3 | 2 | 4 | 3 | 2 | 3 | 3 | 2 | 2
C8- - C8+ | C8 | 2 | 2 | 3 | 3 | 2 | 3 | 2 |   | 3 | 3 | 4 | 2 | 3 | 2 | 3 | 3 | 2 | 2
C9- - C9+ | C9 | 3 | 3 | 2 | 2 | 3 | 2 | 3 | 3 |   | 4 | 3 | 3 | 2 | 3 | 2 | 2 | 3 | 3
C10- - C10+ | C10 | 3 | 3 | 2 | 2 | 3 | 2 | 3 | 3 | 4 |   | 3 | 3 | 2 | 3 | 2 | 2 | 3 | 3
C11- - C11+ | C11 | 2 | 2 | 3 | 3 | 2 | 3 | 2 | 4 | 3 | 3 |   | 2 | 3 | 2 | 3 | 3 | 2 | 2
C12- - C12+ | C12 | 2 | 2 | 3 | 3 | 2 | 3 | 4 | 2 | 3 | 3 | 2 |   | 3 | 2 | 3 | 3 | 2 | 2
C13- - C13+ | C13 | 3 | 3 | 2 | 2 | 3 | 4 | 3 | 3 | 2 | 2 | 3 | 3 |   | 3 | 2 | 2 | 3 | 3
C14- - C14+ | C14 | 2 | 2 | 3 | 3 | 4 | 3 | 2 | 2 | 3 | 3 | 2 | 2 | 3 |   | 3 | 3 | 2 | 2
C15- - C15+ | C15 | 3 | 3 | 2 | 4 | 3 | 2 | 3 | 3 | 2 | 2 | 3 | 3 | 2 | 3 |   | 2 | 3 | 3
C16- - C16+ | C16 | 3 | 3 | 4 | 2 | 3 | 2 | 3 | 3 | 2 | 2 | 3 | 3 | 2 | 3 | 2 |   | 3 | 3
C17- - C17+ | C17 | 4 | 4 | 3 | 3 | 2 | 3 | 2 | 2 | 3 | 3 | 2 | 2 | 3 | 2 | 3 | 3 |   | 4
C33- - C33+ | C18 | 4 | 4 | 3 | 3 | 2 | 3 | 2 | 2 | 3 | 3 | 2 | 2 | 3 | 2 | 3 | 3 | 4 |  

</body>

</html>

**Comment by markheckmann** [comment]
Hi Sami, thanks for reporting the bug! Is that output from the resulting Excel file? Can you please provide the raw data as well? Also, what settings did you use for the analysis?

**Comment by sambrhim** [comment]
Hallo Mark,

im Anhang findest du ein RStudio-Projekt, das alle Informationen enthält. Ein Minimalbeispiel mit library(reprex) habe ich nicht erstellen können. Für weitere Fragen stehe ich gerne zur Verfügung und würde mich freuen, mit Dir persönlich zu sprechen. 

Liebe Grüße 

Sami 

[20220404si2mh.zip](https://github.com/markheckmann/OpenRepGrid.ic/files/8411694/20220404si2mh.zip)

**Comment by markheckmann** [comment]
Hi Sami,  

I had a look at the results and your remarks. 

1. The results are correct, however
2. maybe slightly misleading in the way they are presented.

**1. Example: constructs `C1` vs. `C18`**

![image](https://user-images.githubusercontent.com/152927/164974901-d5e775b9-1e97-45a1-b826-2acb50a0c3ea.png)

The constructs C1 and C18 have no matches the way they are oriented. However, if one of the constructs is reversed, they have 4 matches and this a perfect inverse or negative relation. In the Excel sheet `calculations`, the result matrix named `Number of matches between constructs` counts the number of matches in either way, i.e. also with one of the constructs reversed, as constructs could be negatively related. In other words, it orients the constructs in a way, to maximize their relatedness. To indicate this, there is a negative entry at the corresponding position in the 2nd result matrix `Relatedness by criterion (matches >= 3)`. In other words: The constructs `C1` vs. `C18` have 4 matches and a perfect *negative* relation, when on construct is reversed. 

![image](https://user-images.githubusercontent.com/152927/164975662-b6c3f3cb-63c8-40db-8ebe-6ab1e5d3835a.png)


**2) Presentation of results in Excel file**

I concede that this is not obvious at first glance. I will add an explanation inside the Excel ouput to better explain this. 
There is, however, an explanation in the method section/tab of the software:

>  It is important to recognise that constructs may be negatively related, such that the preferred pole of one construct is related to the non-preferred pole of another. It is important to capture these negative relationships as well as the positive ones, since these can indication tensions and contradictions within the person’s construing. So in inspecting the construct pairs you should also identify any pair where there were instances of this (ie a tick against an element for one construct and a cross for the other). In the case of our research, we looked for cases where there were 5 or more mis-matches, ie. 2 or fewer ‘matches’. 

So basically, the matrix as shown in the output reflects the way the analysis has been implented technically. Instead of using a low *and* high number of matches to indicate relatedness, in my implementation the maximal number of matches under optional reversal is used. The result is the same, but the implementation was easier for me, as I recall. Yet, I agree that this deviates from the description in the methods section. I will add this as a future improvement issue.

HTH
Mark

**Comment by markheckmann** [comment]
In the next release version `v0.6.0` (coming soon), there will an additional matrix that more closely corresponds to the method section and is like the one you expected. Here is a preview of the output.

![image](https://user-images.githubusercontent.com/152927/164977762-56418235-7a6d-4eb7-b325-57e4c86b5412.png)

**Comment by markheckmann** [comment]
One hint: you can create the Excel file output using your file without using the shiny UI as follows (see package docs `?OpenRepGrid.ic`).
```R
library(tidyverse)
library(openxlsx)
library(igraph)
library(OpenRepGrid.ic)

file <-"kgw.xlsx"
x <- read.xlsx(file)           
tests <- check_excel_input(x)   
l <- network_graph_images(x, min_clique_size = 3, show_edges = TRUE, min_matches = 3)    
file_tmp <- create_excel_output(file, l)      
file.show(file_tmp)
```

**Comment by sambrhim** [comment]
Lieber Mark,

ich bin erst jetzt dazu gekommen, Dir zu antworten. Vielen Dank für Deine ausführliche Erklärung und den Hinweis, wie ich die Excel-Dateiausgabe mit R-Syntax erstellen kann.

Liebe Grüße 

Sami

PS: Übrigens die Art und Weise, wie Du es erklärst, ist ganz nach meinem Geschmack :)
